app/workers/recover_lost_jobs_worker.rb

```diff
@@ -2,5 +2,18 @@ class RecoverLostJobsWorker < ApplicationWorker
   def perform
     RecoverLostJobsWorker::UnprocessedCheck.new.call
     RecoverLostJobsWorker::MissingDigestRunsCheck.new.call
+    recover_old_pending_emails
```

Comment:
Could you explain the decision to put the code in the worker?

Reply:
Sure thing - I initially popped it in the `UnprocessedCheck` class but felt it didn't really sit right as the email recovery code is quite different (batches, different fields compared and different worker call method), my main reasons were:

* The class is called `Unprocessed*` and all recovery inside the class focuses on the `processed_at` field (makes sense)

* Pending email recovery can't share the methods used in all other recovery attempts so it just felt like a bespoke bolt on

As a benefit of doing this we've added a spec for the worker in the process which tests the super simple class calls which is similar to our approach when [collecting metrics](https://github.com/alphagov/email-alert-api/blob/master/spec/workers/metrics_collection_worker_spec.rb).

Reply:
I agree the `UnprocessedCheck` is doing something a bit different to what we need here, and those two points are strong arguments that it's a separate concern, like missing digest runs, which we've encapsulated in a separate class. What I'm confused about is why the code to recover emails is inconsistent with how we've structured the other recovery code. Why not put it in its own class?

Also, I think adding tests to the existing worker class is unrelated to this change, so we could always get the benefit of that separately.

Reply:
I didn't think it warranted it's own class because the recovery of pending emails is simpler than the recovery code contained within `UnprocessedCheck` (multiple models) and `MissingDigestRunsCheck` (more complicated logic).

If you feel strongly I can create its own class.

Reply:
I do feel strongly, but I think this is more than subjective: while complexity is indeed an argument for what merits a class, consistency is also a strong argument. If I'm reading this code for the first time, then I'm faced with:

- A new concept called "Check", with code in two classes.
- A method, which says it does recovery.

The "Check" concept and the method actually represent the same thing. Conversely, the purpose of the worker is to recovery "everything", but its private methods are specific to one kind of recovery.

If I'm adding a new check in the future, it's also unclear whether I should put it in worker directly, or in its own class, as there's now a precedent for both.

Reply:
Hmm I see your point. 

I did think about naming the method in respect to the `*Check` classes which may have been easier to parse, e.g. `old_pending_emails_check` but decided against it as it is not really a 'check' as the method is actively recovering work. 

The more I think about the more I wonder the benefit of having these class suffixed with 'Check' as it's a little confusing as they do more than just check something. However, I've added the class as requested so hopefully this should be ready to go!

Reply:
Thanks. Yeah, I see what. you mean with `Check`. We could probably do just lose the suffix, as it was just a subjective choice of how to describe the classes, which are arguably described by their parent class.